plugins/nordic-hid/fu-nordic-hid-archive.c

```diff
@@ -119,53 +262,42 @@ fu_nordic_hid_archive_parse(FuFirmware *firmware,
 		if (blob == NULL)
 			return FALSE;
 
-		if (json_object_has_member(obj, "version_B0")) {
-			bootloader_name = "B0";
-			image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_B0, NULL);
-		} else if (json_object_has_member(obj, "version_MCUBOOT")) {
-			bootloader_name = "MCUBOOT";
-			image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_MCUBOOT, NULL);
-		} else if (json_object_has_member(obj, "version_MCUBOOT+XIP")) {
-			bootloader_name = "MCUBOOT+XIP";
-			image = g_object_new(FU_TYPE_NORDIC_HID_FIRMWARE_MCUBOOT, NULL);
-		} else {
-			g_set_error_literal(error,
-					    FWUPD_ERROR,
-					    FWUPD_ERROR_INVALID_FILE,
-					    "only B0 and MCUboot bootloaders are supported");
+		if (!fu_nordic_hid_archive_parse_file_image_create(obj,
+								   &image,
+								   &bootloader_name,
+								   error))
```

Comment:
i'm not convinced splitting this out makes it any clearer

Reply:
Ok, I will rework this part a bit (split getting bootloader name from `image` creation